test/tst_cosimulate.cpp

```diff
@@ -97,23 +98,29 @@ private slots:
 };
 
 void tst_Cosimulate::trapHandler() {
-  unsigned status = ProcessorHandler::get()->getProcessor()->getRegister(
-      RegisterFileType::GPR,
-      ProcessorHandler::get()->getProcessor()->implementsISA()->syscallReg());
-
-  /// @todo: Generalize this by having ISA report exit syscall codes
-  if (status == RVABI::SysCall::Exit || status == RVABI::SysCall::Exit2) {
-    m_stop = true;
+  if (auto reg = ProcessorHandler::get()
+                     ->getProcessor()
+                     ->implementsISA()
+                     ->syscallReg();
+      reg.has_value()) {
```

Comment:
Nit: Generally, if-pyramids should be avoided; allows for easier to read code (and more optimized due to shorter branch distances). Instead, inverted if conditions and early returns are preferred. 

That is, this function could be rewritten as:
```c++
void tst_Cosimulate::trapHandler() {
  auto reg = ProcessorHandler::get()
                      ->getProcessor()
                      ->implementsISA()
                      ->syscallReg();
  if(!reg.has_value())
    return;
  
  unsigned status = ...
}
```

Also... should this be an assert? This would be a pretty wonky test if there wasn't a syscall register available x).